lib/datadog/open_feature/native_evaluator.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'json'
+require_relative 'ext'
+require_relative 'resolution_details'
+
+module Datadog
+  module OpenFeature
+    # Evaluation using native extension
+    class NativeEvaluator
+      def initialize(configuration)
+        @libdatadog_api_failure = Datadog::Core::LIBDATADOG_API_FAILURE
+        if configuration && !@libdatadog_api_failure
+          @configuration = Datadog::Core::FeatureFlags::Configuration.new(configuration)
+        end
+      end
```

Comment:
This will be done at component level and we would not enable component if libdatadog is not loaded. https://github.com/DataDog/dd-trace-rb/pull/5047/files#diff-49253ad22909b3e9d6d6ccbde589b945d565e50dd33078addf02dd12c3069c06R35-R40

I think we can assume that it's loaded and working and will not be called otherwise.